assets/javascripts/discourse/initializers/discourse-calendar.js

```diff
@@ -663,6 +779,49 @@ function initializeDiscourseCalendar(api) {
     }
   }
 
+  function _setTimezoneOffset(info) {
+    if (
+      !siteSettings.enable_timezone_offset_for_calendar_events ||
+      info.view.type === "listNextYear"
+    ) {
+      return;
+    }
+
+    const timezoneOffset = info.event.extendedProps.timezoneOffset;
+    const segmentDuration = info.el.parentNode?.colSpan;
+
+    const basePctOffset = 100 / segmentDuration;
+    const basePxOffset = 5.5 - (segmentDuration - 1) * 0.78;
```

Comment:
This is a little bit too magic. Maybe add a comment explaining the reasoning for this?